src/pathfinder/yapf/yapf_ship.cpp

```diff
@@ -90,16 +90,26 @@ class CYapfFollowShipT
 
 		Node *pNode = pf.GetBestNode();
 		if (pNode != NULL) {
+			uint steps = 0;
+			for (Node *n = pNode; n->m_parent != NULL; n = n->m_parent) steps++;
+
 			/* walk through the path back to the origin */
 			Node *pPrevNode = NULL;
 			while (pNode->m_parent != NULL) {
+				if (steps > 1 && --steps < YAPF_SHIP_PATH_CACHE_LENGTH) {
+					TrackdirByte td;
```

Comment:
The path is stored even if it doesn't find a valid path to destination. This may lead to problems in ship_cmp (see comment on that file).

Reply:
This is right, but I was trying to explain that on ship_cmd there might be an issue. Storing the path is no problem. But storing the best path doesn't mean a path to destination is found (see ship_cmd comment).